lib/resque/unique_in_queue/queue.rb

```diff
@@ -4,14 +4,20 @@ module Queue
       def queued?(queue, item)
         return false unless is_unique?(item)
 
-        redis.get(unique_key(queue, item)) == '1'
+        key   = unique_key(queue, item)
+        value = unique_value(queue, item)
+
+        redis.sismember(key, value)
```

Comment:
Don't you have to compare the result with `1`?

https://redis.io/commands/sismember#return-value

> Return value
> Integer reply, specifically:
> 
> 1 if the element is a member of the set.
> 0 if the element is not a member of the set, or if key does not exist.